[DOCS] New option for users to run query packs from alert (#2249)
elastic/security-docs: docs/detections/alerts-run-osquery.asciidoc

* Verify that {fleet-guide}/view-elastic-agent-status.html[{agent}'s status] is *Healthy*. Refer to {fleet-guide}/fleet-troubleshooting.html[{fleet} Troubleshooting] if it is not.
============

[float]
[[osquery-alert-action]]
=== Run Osquery

. Do one of the following from the Alerts table:
** Click the *View details* button to open the Alert details flyout, then click *Take action -> Run Osquery*.
** Select the *More actions* menu (*...*), then select *Run Osquery*.
. Choose to run a single query or a query pack.
. Select one or more {agent}s or groups to query. Start typing in the search field to get suggestions for {agent}s by name, ID, platform, and policy.

+
NOTE: The host associated with the alert is automatically selected. You can specify additional hosts to query.

. Specify the query or pack to run:

> ** *Query*: Select a saved query or enter a new one in the text box beneath. After you enter the query, you can expand the **Advanced** section to view or set {kibana-ref}/osquery.html#osquery-map-fields[mapped ECS fields] included in the results from the live query. Mapping ECS fields is optional.

Author: Oops, I thought I deleted this. Agree with removing the word.

** *Pack*: Select from query packs that have been loaded and activated. After you select a pack, all of the queries in the pack are displayed.
+

> TIP: Refer to {kibana-ref}/osquery.html#osquery-prebuilt-packs-queries[prebuilt packs] to learn about using and managing Elastic prebuilt packs.

Reviewer: 👍

+
[role="screenshot"]
image::images/setup-query.png[width=80%][height=80%][Shows how to set up a single query]

. Click **Submit**. Queries will timeout after 5 minutes if there are no responses.
+
TIP: To save the query for future use, click *Save for later* and define the ID,
description, and other {kibana-ref}/osquery.html#osquery-manage-query[details].

[float]
[[osquery-results-single]]
=== Review single query results

> Results for single queries are displayed in the *Results* tab. The number of agents that were queried and their responses are shown in the top right corner of the results table. Agent responses can be `Sucessful`, `Not yet responded` (pending), and `Failed`.

Author: Hm, this is odd. @patrykkopycinski @james-elastic were the agent statuses removed from the results table in a recent release? I ran a single query from Kibana and Security to see if the table designs were different and I'm not able to see the agent results in either areas.

> Results for single queries are displayed in the *Results* tab. When you run a query, the number of agents queried and query status temporarily display in a status bar above the results table. Agent responses can be `Sucessful`, `Not yet responded` (pending), and `Failed`.

Reviewer 2: ```suggestion
Results for single queries appear in the *Results* tab. When you run a query, the number of agents queried and query status temporarily display in a status bar above the results table. Agent responses can be `Sucessful`, `Not yet responded` (pending), and `Failed`.
```

[role="screenshot"]
image::images/query-results.png[width=80%][height=80%][Shows query results]

[float]
[[osquery-results-pack]]
=== Review query pack results

> Results for each query in the pack are displayed in the *Results* tab. Click the expand button (image:images/pack-expand-button-osquery.png[Click markdown icon,20,20]) at the far right of each query row to display query results. The number of agents that were queried and their responses are shown for each query. Agent responses are color-coded. Green is `Sucessful`, `Not yet responded` (pending) is gray, and `Failed` is red.

Author: The agent responses and results are included, they're just shown inline, as opposed to above the results table. 
![agent-results-inline](https://user-images.githubusercontent.com/79124755/184447648-128042e6-0d83-4547-a6d1-b1b141b720e4.gif)

> Results for each query in the pack are displayed in the *Results* tab. Click the expand button (image:images/pack-expand-button-osquery.png[Click markdown icon,20,20]) at the far right of each query row to display query results. The number of agents that were queried and their responses are shown for each query. Agent responses are color-coded. Green is `Sucessful`, `Not yet responded` (pending) is gray, and `Failed` is red.

Reviewer 2: ```suggestion
Results for each query in the pack appear in the *Results* tab. Click the expand button (image:images/pack-expand-button-osquery.png[Click markdown icon,20,20]) at the far right of each query row to display query results. The number of agents that were queried and their responses are shown for each query. Agent responses are color-coded. Green is `Sucessful`, `Not yet responded` (pending) is gray, and `Failed` is red.
```

[role="screenshot"]
image::images/pack-query-results.png[width=80%][height=80%][Shows query results]

[float]
[[osquery-investigate]]
=== Investigate query results

From the results table, you can:

* Click the *View in Discover* button (image:images/discover-button-osquery.png[Click markdown icon,20,20])  to explore the results in Discover.
* Click the *View in Lens* button (image:images/lens-button-osquery.png[Click markdown icon,20,20])  navigate to Lens, where you can use the drag-and-drop *Lens* editor to create visualizations.
* Click the *Timeline* button (image:images/timeline-button-osquery.png[Click markdown icon,20,20]) to investigate a single query result in Timeline or *Add to timeline investigation* to investigate all results. This option is only available for single query results

+

> TIP: An `action_ID` is generated when you run an Osquery query. The `action_ID` field and value pair is passed to the Timeline's KQL filter when you select the option to open all results in Timeline.

Reviewer 3: I know this isn't part of the PR proper, but it's unclear what exactly the user would do with this info. I think it gets bogged down in explaining the mechanics of an action_ID instead of focusing on its usefulness.

Here's one possibility, which I don't think needs to be highlighted as a Tip or even a Note, because it's really just basic information; it could be just another sentence added to the previous bullet/para about Timeline:
```suggestion
When you open all results in Timeline, the events in Timeline are filtered based on the `action_ID` generated by the Osquery query.
```

Author: I see the point you're making and your revision is correct. :)
